src/main/scala/io/iohk/praos/domain/VirtualGenesis.scala

```diff
@@ -0,0 +1,37 @@
+package io.iohk.praos.domain
+import io.iohk.praos.util.Logger
+
+trait VirtualGenesis {
```

Comment:
Maybe it would be better to rename this `EpochGenesisCalculator`?

Reply:
Agree